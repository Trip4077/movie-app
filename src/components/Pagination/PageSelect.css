/*
---------------------------Pagination Styles---------------------------
*/

.pagination {
    display: flex;
    justify-content: space-around;
    align-items: center;
  
    width: 75%;
    margin: 30px auto;
  }
  
  .pagination li {
    margin: 5px;
    height: 30px;
    width: 30px;
  
    font-size: 18px;
    font-weight: 500px;
  }
  
  .pagination li:hover {
    background-color: darkorange;
    color: white;
  }