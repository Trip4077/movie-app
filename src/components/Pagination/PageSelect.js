import React from 'react';

const PageSelect = props => {
    
    const selectPage = e => {
        props.performSearch(e, Number(e.target.innerText))
    }

    const incrementPage = e => {
        props.performSearch(e, (props.page + 1))
    }

    const decrementPage = e => {
        props.performSearch(e, (props.page - 1))
    }

    return(
        <div className='page-index'>
            <ul className="pagination" data-testid="page-select">
                <li className="waves-effect" onClick={decrementPage}><i className="material-icons">chevron_left</i></li>

                { props.pageList ? props.pageList.map(pageNum => <li key={pageNum} onClick={selectPage} className="waves-effect">{pageNum}</li>) 
                                        : <span></span> }

                <li className="waves-effect" onClick={incrementPage}><i className="material-icons">chevron_right</i></li>
            </ul>
        </div>
    );
}

export default PageSelect;