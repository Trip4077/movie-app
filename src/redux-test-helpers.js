import React from 'react';

import { createStore } from 'redux';
import { Provider } from 'react-redux';

import { MemoryRouter } from 'react-router';

import { render } from 'react-testing-library';

import * as ACTIONS from './actions';

function reducer(state = { userReducer: { user: { id: 1 }}}, action) {
    switch(action.type) {
        case ACTIONS.LOADING:
            return {
                ...state,
                loading: true
            }

        case ACTIONS.END_LOAD:
            return {
                ...state,
                loading: false
            }

        case ACTIONS.UPDATE_FAV_LIST:
            return {
                loading: false,
                favorites: action.payload,
                schedule: state.schedule
            }

        case ACTIONS.GET_INFO:
            return {
                ...state,
                loading: false,
                info: action.payload
            }

        case ACTIONS.UPDATE_SCHEDULE:
            return {
                loading: false,
                favorites: state.favorites,
                schedule: action.payload
            }

        case ACTIONS.TEXT_SENT:
            return {
                ...state,
                loading: false
            }

            case ACTIONS.REGISTER: 
            return {
                loading: false,
                userID: action.payload
            }

        case ACTIONS.LOGIN:
            localStorage.setItem( 'token', action.payload.token )

            delete action.payload.user.password

            return {
                loading: false,
                user: action.payload.user
            }

        case ACTIONS.LOGOUT:
            return { ...initialState }

        default:
            return {
                ...state
            }
    }
}

export function renderWithRedux( 
    ui, 
    { initialState, store = createStore(reducer, initialState) } = {} ) {
        return {
            ...render( <Provider store={store}>
                         <MemoryRouter>
                            <div> {ui} </div>
                         </MemoryRouter>
                       </Provider> ),
            store
        }
    };