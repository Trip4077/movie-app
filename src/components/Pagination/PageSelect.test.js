import React from 'react';

import { cleanup, fireEvent } from 'react-testing-library';
import { renderWithRedux } from '../../redux-test-helpers';

import apple from 'react-test-renderer';

import PageSelect from './PageSelect';

afterEach(cleanup);

describe('<PageSelect />', () => {

  it('renders without crashing', () => {
    renderWithRedux(<PageSelect />);
  });

  it('renders unordered list of <li> elements', () => {
    const { getByTestId } = renderWithRedux(<PageSelect pageList={[ 1 ]} />);

    getByTestId('page-select');
  });

  it('renders 2 + pageList\'s length <li> elements', () => {
    const { getByTestId } = renderWithRedux(<PageSelect pageList={[ 1, 2 ]} />);

    const list = getByTestId('page-select');

    expect(list.childNodes.length).toBe(4);
  });

  
  it('renders <li> text to be page number', () =>{
    const { getByTestId } = renderWithRedux(<PageSelect pageList={[ 1, 2, 3, 4 ]} />);

    const list = getByTestId('page-select');
    const li = list.childNodes;

    expect(li[1].textContent).toBe('1');
    expect(li[2].textContent).toBe('2');
    expect(li[3].textContent).toBe('3');
    expect(li[4].textContent).toBe('4');    
  });

  it('fires props function when <li> elements are clicked', () => {
      const mock = jest.fn();
      const { getByTestId } = renderWithRedux(<PageSelect pageList={[ 1, 2, 3 ]} performSearch={mock} />);

      const list = getByTestId('page-select');
      const li = list.childNodes;

      fireEvent.click(li[0]);
      expect(mock).toHaveBeenCalled();

      fireEvent.click(li[2]);
      expect(mock).toHaveBeenCalledTimes(2);

      fireEvent.click(li[4]);
      expect(mock).toHaveBeenCalledTimes(3);
  });

  it('should match snapshot - default', () => {
      const tree = apple.create(<PageSelect />);

      expect(tree.toJSON()).toMatchSnapshot();
  });

  it('should match snapshot - with props', () => {
    const tree = apple.create(<PageSelect pageList={[ 1, 2, 3, 4, 5]} />);

    expect(tree.toJSON()).toMatchSnapshot();
  });
});