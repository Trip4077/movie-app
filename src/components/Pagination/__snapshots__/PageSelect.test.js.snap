// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageSelect /> should match snapshot - default 1`] = `
<div
  className="page-index"
>
  <ul
    className="pagination"
    data-testid="page-select"
  >
    <li
      className="waves-effect"
      onClick={[Function]}
    >
      <i
        className="material-icons"
      >
        chevron_left
      </i>
    </li>
    <span />
    <li
      className="waves-effect"
      onClick={[Function]}
    >
      <i
        className="material-icons"
      >
        chevron_right
      </i>
    </li>
  </ul>
</div>
`;

exports[`<PageSelect /> should match snapshot - with props 1`] = `
<div
  className="page-index"
>
  <ul
    className="pagination"
    data-testid="page-select"
  >
    <li
      className="waves-effect"
      onClick={[Function]}
    >
      <i
        className="material-icons"
      >
        chevron_left
      </i>
    </li>
    <li
      className="waves-effect"
      onClick={[Function]}
    >
      1
    </li>
    <li
      className="waves-effect"
      onClick={[Function]}
    >
      2
    </li>
    <li
      className="waves-effect"
      onClick={[Function]}
    >
      3
    </li>
    <li
      className="waves-effect"
      onClick={[Function]}
    >
      4
    </li>
    <li
      className="waves-effect"
      onClick={[Function]}
    >
      5
    </li>
    <li
      className="waves-effect"
      onClick={[Function]}
    >
      <i
        className="material-icons"
      >
        chevron_right
      </i>
    </li>
  </ul>
</div>
`;
